File: app/src/main/java/com/mytlogos/enterprise/background/room/model/RoomSimpleRelease.java
package com.mytlogos.enterprise.background.room.model;

public class RoomSimpleRelease {
    public final int partId;
    public final int episodeId;
    public final String url;

    public RoomSimpleRelease(int partId, int episodeId, String url) {
        this.partId = partId;
        this.episodeId = episodeId;
        this.url = url;
    }

    @Override
    public String toString() {
        return "RoomSimpleRelease{" +
                "partId=" + partId +
                ", episodeId=" + episodeId +
                ", url='" + url + '\'' +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        RoomSimpleRelease that = (RoomSimpleRelease) o;

        if (partId != that.partId) return false;
        if (episodeId != that.episodeId) return false;
        return url.equals(that.url);
    }

    @Override
    public int hashCode() {
        int result = partId;
        result = 31 * result + episodeId;
        result = 31 * result + url.hashCode();
        return result;
    }
}

File: app/src/main/java/com/mytlogos/enterprise/background/room/model/RoomSimpleRelease.kt
package com.mytlogos.enterprise.background.room.model

class RoomSimpleRelease(val partId: Int, val episodeId: Int, val url: String) {
    override fun toString(): String {
        return "RoomSimpleRelease{" +
                "partId=" + partId +
                ", episodeId=" + episodeId +
                ", url='" + url + '\'' +
                '}'
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val that = o as RoomSimpleRelease
        if (partId != that.partId) return false
        return if (episodeId != that.episodeId) false else url == that.url
    }

    override fun hashCode(): Int {
        var result = partId
        result = 31 * result + episodeId
        result = 31 * result + url.hashCode()
        return result
    }
}